src/OpenTelemetry.AutoInstrumentation.Native/util.cpp

```diff
@@ -97,6 +99,53 @@ std::vector<WSTRING> GetEnvironmentValues(const WSTRING& name)
     return GetEnvironmentValues(name, L',');
 }
 
+
+std::vector<WSTRING> GetEnvironmentVariables(const std::vector<WSTRING> &prefixes)
+{
+    std::vector<WSTRING> env_strings;
+#ifdef _WIN32
+    auto env_variables = GetEnvironmentStrings();
+    int prev = 0;
+    for (int i = 0;; i++) {
+        if (env_variables[i] == '\0')
+        {
+            auto env_variable = WSTRING(env_variables + prev, env_variables + i);
+            for (const auto& prefix : prefixes)
+            {
+                if (env_variable.find(prefix) == 0)
+                {
+                  env_strings.push_back(env_variable);
+                  break;
+                }
+            }
+
+            prev = i + 1;
+            if (env_variables[i + 1] == '\0')
+            {
+                break;
+            }
+        }
+    }
+
+    FreeEnvironmentStrings(env_variables);
+#else
+    const auto env_variables = environ;
+    for (char** current = environ; *current; current++)
```

Comment:
If removing the unused above then add the proper const to `current` here.

Reply:
It is not possible as current is incremented in last part of for loop: `current++`